benchmark/src/util.ts

```diff
@@ -0,0 +1,85 @@
+import { BenchmarkFunction } from "./benchmark_types";
+
+export function round(num: number, decimalPlaces: number = 0) {
+    return tonumber(string.format(`%.${decimalPlaces}f`, num));
+}
+
+export const json: {
+    decode: (this: void, str: string) => {};
+    encode: (this: void, val: any) => string;
+} = require("json");
+
+export function readFile(path: string): string | undefined {
+    const fileOpenArray = io.open(path, "rb");
+
+    if (fileOpenArray && fileOpenArray[0]) {
+        const fileHandle = fileOpenArray[0];
+        let fileContent = readAll(fileHandle);
+        fileHandle.close();
+
+        if (fileContent) {
+            return fileContent;
+        }
+    }
+}
+
+export function readAll(file: LuaFile): string | undefined {
+    let content: (string | undefined)[];
+    if (_VERSION == "Lua 5.3") {
+        // @ts-ignore
+        content = file.read("a");
+    } else {
+        // JIT
+        // @ts-ignore
+        content = file.read("*a");
+    }
+
+    if (content && content[0]) {
+        return content[0];
+    }
+}
+
+export function readDir(dir = ""): string[] | undefined {
+    let isWindows = false;
+    let [success, findHandle] = pcall(() => io.popen(`find '${dir}' -maxdepth 1 -type f | sed "s|^\./||"`));
+
+    if (!success) {
+        [success, findHandle] = pcall(() => io.popen(`dir /A-D /B ${dir}`));
+        isWindows = true;
+    }
+
+    if (success) {
+        // appereantly TS can't infer this on it's own
+        findHandle = findHandle as LuaFile;
+
+        const findResult = readAll(findHandle);
+        findHandle.close();
+
+        if (findResult) {
+            let files = findResult.split("\n");
+            if (isWindows) {
```

Comment:
Maybe instead we could make `find` return just file names with `-printf "%f\n"`, so we can have a single code path for both platforms

Reply:
`-printf` is a rather new unix extension to `find` and not part of the posix standard. So it won't work on BSD systems (macOS) and older unix systems.